### ISSUE TITLE
In TLS case NTLM is not handled.

### ISSUE THREAD

**Issue by fifafu** [tediousjs/tedious#338]
If you look at this (in connection.js):
```
SENT_TLSSSLNEGOTIATION: {
    name: 'SentTLSSSLNegotiation',
    events: {
      socketError: function() {
        return this.transitionTo(this.STATE.FINAL);
      },
      connectTimeout: function() {
        return this.transitionTo(this.STATE.FINAL);
      },
      data: function(data) {
        return this.messageIo.tlsHandshakeData(data);
      },
      message: function() {
        if (this.messageIo.tlsNegotiationComplete) {
          this.sendLogin7Packet();
          return this.transitionTo(this.STATE.SENT_LOGIN7_WITH_STANDARD_LOGIN);
        }
      }
    }
  },
```
Only standard login is handled. 

It would be good to handle the encrypted case for NTLM too, maybe like this:

```
if (this.config.domain) {
          return this.transitionTo(this.STATE.SENT_LOGIN7_WITH_NTLM);
        } else {
          return this.transitionTo(this.STATE.SENT_LOGIN7_WITH_STANDARD_LOGIN);
        }
```

**Comment by arthurschreiber** [comment]
You're right, Encryption and NTLM authentication don't work together, and your patch seems to be correct. Would you mind opening a PR with your patch and accompanying test cases?

**Comment by arthurschreiber** [comment]
This was fixed with #367.
